flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GroupAggProcessFunction.scala

```diff
@@ -118,19 +131,33 @@ class GroupAggProcessFunction(
       }
     }
 
-    state.update(accumulators)
+    if (inputCnt > 0) {
```

Comment:
I'm not sure I understand the issue. Can you add a commit with the changes you propose and a test case the validates the changes (i.e., a query that would not produce the correct result without your changes)?